internal/services/gcp.go

```diff
@@ -0,0 +1,83 @@
+package services
+
+import (
+    "cloud.google.com/go/storage"
+    "context"
+    "encoding/json"
+    "errors"
+    "expo-open-ota/config"
+    "fmt"
+    "os"
+    "sync"
+    "time"
+)
+
+var (
+    gcsClient     *storage.Client
+    initGCSClient sync.Once
+)
+
+func GetGCSClient() (*storage.Client, error) {
+    var err error
+    initGCSClient.Do(func() {
+        ctx := context.Background()
+        gcsClient, err = storage.NewClient(ctx)
+    })
+    if err != nil {
+        return nil, fmt.Errorf("error initializing GCS client: %w", err)
+    }
+    return gcsClient, nil
+}
+
+type googleCreds struct {
+    ClientEmail string `json:"client_email"`
+    PrivateKey  string `json:"private_key"`
+}
+
+func loadGoogleCreds() (*googleCreds, error) {
+    path := os.Getenv("GOOGLE_APPLICATION_CREDENTIALS")
+    if path == "" {
+        return nil, errors.New("GOOGLE_APPLICATION_CREDENTIALS not set")
+    }
+    b, err := os.ReadFile(path)
+    if err != nil {
+        return nil, fmt.Errorf("read GOOGLE_APPLICATION_CREDENTIALS: %w", err)
+    }
```

Comment:
Should only `credentials.json` be supported? Maybe a JSON in env should be also considered?